### PR TITLE
Add io_config, dummy_inputs_func and dynamic axes to PytorchModel

### PR DESCRIPTION
**Pull request by jambayk**
Currently, model specific information such as `input_names`, `input_shapes`, etc. are provided as config parameters for passes like `ONNXConversion`. We want to support composite models which are made of multiple models. For such cases, we cannot provide the io information to a pass for all at once. 

So, we make the io information optional init parameters of PytorchModel. The passes will in turn get them from the model itself. This also makes more sense conceptually since these are attributes of the model and not the pass. 

We will make similar changes to the ONNXModel and onnx related passes in the future. Any model specific information should be part of the model and not the pass config.

### CODE REVIEW

File: olive/model.py [hunk 1]
```diff
@@ -362,13 +415,49 @@ def load_model(self, rank: int = None) -> torch.nn.Module:
     def prepare_session(self, inference_settings: Dict[str, Any], device: Device, rank: int = None):
         return self.load_model().eval()
 
+    def get_dummy_inputs(self):
```
Comment (guotuofeng) [inline olive/model.py:418]: these logic seems not be related to PyTorch model. Instead, it should be part of dataset.

**Review by guotuofeng — CHANGES_REQUESTED** [review]
suggest to move the dummy data generation to dummy dataset like https://github.com/microsoft/Olive/blob/main/olive/evaluator/evaluation.py#L354

File: olive/model.py (hunk 1 above)
Comment (jambayk) [reply olive/model.py:418]: It is part of pytorch model. When converting to onnx or during qat, we need dummy input.

Comment (jambayk) [reply olive/model.py:418]: Can you please unblock this PR? This is urgent for whisper model support.

Comment (jambayk) [reply olive/model.py:422]: Added TODO as discussed offline.

File: examples/bert_ptq_cpu/docker/Dockerfile [hunk 2]
```diff
@@ -6,6 +6,6 @@ RUN pip install azure-ai-ml \
             datasets \
             transformers \
             onnxconverter_common \
-            git+https://github.com/microsoft/Olive.git
+            git+https://github.com/microsoft/Olive.git@jambayk/model-info
```
Comment (trajepl) [inline examples/bert_ptq_cpu/docker/Dockerfile:9]: Please remember to remove it.

**Review by trajepl — APPROVED** [review]

**Review by guotuofeng — APPROVED** [review]

**Pull request merged**